components/datetime/src/mock/timezone.rs

```diff
@@ -0,0 +1,297 @@
+    Off,
+}
+
+/// The GMT offset in seconds for a `ZonedDateTime`.
+#[derive(Copy, Clone, Debug, Default)]
+pub struct GmtOffset(i32);
+
+/// Get the ascii byte of a numeric digit.
+/// Only called on digits from in range 0..=9.
+const fn ascii_digit(digit: u8) -> u8 {
+    digit + b'0'
+}
+
+/// Formats a time segment with optional zero padding.
+const fn format_segment(n: u8, padding: ZeroPadding) -> TinyStr4 {
+    // This section is safe because it operates on a finite set of 0..=60
+    // and it ensures that all TinyStr4s are constructed from valid
+    // little-endian bytes, which is the required internal representation.
+    unsafe {
+        if n >= 10 {
+            TinyStr4::new_unchecked(u32::from_le_bytes([
+                ascii_digit(n / 10),
+                ascii_digit(n % 10),
+                0,
+                0,
+            ]))
+        } else {
+            match padding {
+                ZeroPadding::On => {
+                    TinyStr4::new_unchecked(u32::from_le_bytes([b'0', ascii_digit(n), 0, 0]))
+                }
+                ZeroPadding::Off => {
+                    TinyStr4::new_unchecked(u32::from_le_bytes([ascii_digit(n), 0, 0, 0]))
+                }
+            }
+        }
+    }
+}
+
+impl GmtOffset {
+    /// Formats the hours as a `TinyStr4` with optional zero-padding.
+    pub(crate) fn format_hours(&self, padding: ZeroPadding) -> TinyStr4 {
+        format_segment((self.0 / 3600).abs() as u8, padding)
+    }
+
+    /// Formats the minutes as a `TinyStr4` with zero-padding.
+    pub(crate) fn format_minutes(&self) -> TinyStr4 {
+        format_segment((self.0 % 3600 / 60).abs() as u8, ZeroPadding::On)
+    }
+
+    /// Formats the seconds as a `TinyStr4` with zero-padding.
+    pub(crate) fn format_seconds(&self) -> TinyStr4 {
+        format_segment((self.0 % 3600 % 60).abs() as u8, ZeroPadding::On)
+    }
+
+    /// Whether the GMT offset is positive.
+    pub(crate) fn is_positive(&self) -> bool {
+        self.0 >= 0
+    }
+
+    /// Whether the GMT offset is zero.
+    pub(crate) fn is_zero(&self) -> bool {
+        self.0 == 0
+    }
+
+    /// Whether the GMT offset has non-zero minutes.
+    pub(crate) fn has_minutes(&self) -> bool {
+        self.0 % 3600 / 60 > 0
+    }
+
+    /// Whether the GMT offset has non-zero seconds.
+    pub(crate) fn has_seconds(&self) -> bool {
+        self.0 % 3600 % 60 > 0
+    }
+
+    /// Formats a GMT offset in ISO-8601 format.
+    pub(crate) fn iso8601_format<'a>(
+        self,
+        format: IsoFormat,
+        minutes: IsoMinutes,
+        seconds: IsoSeconds,
+    ) -> Cow<'a, str> {
+        if self.is_zero() && matches!(format, IsoFormat::UtcBasic | IsoFormat::UtcExtended) {
+            return Cow::Owned(String::from("Z"));
+        }
+
+        let extended_format = matches!(format, IsoFormat::Extended | IsoFormat::UtcExtended);
+        let mut s = String::from(if self.is_positive() { '+' } else { '-' });
```

Comment:
So the [ISO-8601 spec](https://www.iso.org/obp/ui/#iso:std:iso:8601:-1:ed-1:v1:en) (section 3.2.4) says to use a minus sign if it's available, and to use the ASCII hyphen as a fallback.

How can we tell if it's available? Querying user fonts sounds like it would be quite the effort just for this.

Reply:
If UTS 35 wants us to use localized digits, then that would imply also using localized +/- symbols.  This is supported in FixedDecimalFormat, which we need to integrate as part of 0.3.

ISO-8601 isn't particularly helpful when it comes to localized formatting.  It is designed for the opposite problem: one universally understood date/time format.

Reply:
Also, I think I'm a little confused about "iso8601_format".  Our job is to render localized time zones.  ISO-8601 isn't localized by definition.  The human-readable version of something like "-05:00" is "GMT-5".  Why do we have a function that produces non-localized results?  Is it required by UTS 35?